### PR TITLE
fix(spacewidget): upgrade from node 14 to node 20

### PR DESCRIPTION
**Pull request by Kesari3008**
Upgraded node 14 to node 20. Updated webpack 4 to webpack 5. 

https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-477304 

Created a new JIRA for fixing the ignored unit tests: https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-558388 

Tests Performed:
1. Selected 'To Person' as the destination type and opened the space. 1:1 message worked. Joining meetings works with audio and video too. Share screen worked too.

Screenshot 2024-09-02 at 6 13 42 PM

2. Selected 'SIP URI' as the destination type, entered the SIP address for a scheduled meeting and was able to join the meeting. Audio and Video worked. Share screen worked too

Screenshot 2024-09-03 at 3 47 46 PM

3. Select 'To Space' as the destination type and open the space. 

Screenshot 2024-09-04 at 3 58 01 AM

4. Select 'To Person ID' as the destination type and open 1:1 space.

### CODE REVIEW

File: scripts/start/commands/samples.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ module.exports = {
   builder: {},
   handler: () => {
     console.info('Starting the samples ...');
-    const command = `webpack-dev-server --config scripts/webpack/webpack.dev.babel.js --hot --inline --history-api-fallback --context ${path.resolve('./samples')}`;
+    const command = `webpack-dev-server --config scripts/webpack/webpack.dev.babel.js --hot --history-api-fallback --context ${path.resolve('./samples')}`;
```
Comment (sreenara) [inline scripts/start/commands/samples.js:11]: What does this `--inline` change in the final command? Could that be the reason for the warnings we are observing?

File: scripts/webpack/webpack.dev.babel.js [hunk 2]
```diff
@@ -16,13 +16,13 @@ const plugins = [
...
     }
   }),
-  new webpack.EnvironmentPlugin([
-    'WEBEX_ACCESS_TOKEN',
-    'WEBEX_CLIENT_ID',
-    'SPACE_ID',
-    'TO_PERSON_EMAIL',
-    'TO_PERSON_ID'
-  ])
+  new webpack.EnvironmentPlugin({
```
Comment (sreenara) [inline scripts/webpack/webpack.dev.babel.js:19]: Why have we made all these entries empty? There was dummy text earlier but now all these are empty strings. Wouldn't that break something?

**Review by sreenara — CHANGES_REQUESTED** [review]
Please see the comments

File: scripts/start/commands/samples.js (hunk 1 above)
Comment (Kesari3008) [reply scripts/start/commands/samples.js:11]: No that's not the reason for warnings, it was due to css-loader version upgrade. This change is required as per webpack 5 config supported

File: scripts/webpack/webpack.dev.babel.js (hunk 2 above)
Comment (Kesari3008) [reply scripts/webpack/webpack.dev.babel.js:19]: It was giving errors so I had to update the value as suggested in the error. But they will only be empty if process.node.env doesn't return anything

File: jest.config.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "roots": [
     "<rootDir>/packages/node_modules/"
   ],
-  "testPathIgnorePatterns": ["<rootDir>/node_modules/"],
+  "testPathIgnorePatterns": ["<rootDir>/node_modules/", "<rootDir>/packages/node_modules/@webex/redux-module-spaces/src/actions.test.js", "<rootDir>/packages/node_modules/@webex/redux-module-users/src/actions.test.js", "<rootDir>/packages/node_modules/@webex/redux-module-media/src/actions.test.js", "<rootDir>/packages/node_modules/@webex/react-redux-spark/src/component.test.js"],
```
Comment (sreenara) [inline jest.config.json:5]: Can we tag the jira we raised for these tests in the comment?

File: .circleci/config.yml [hunk 4]
```diff
@@ -52,10 +52,10 @@ jobs:
           command: npm cache clean --force
       - run:
           name: "Install dependencies"
-          command: npm ci
+          command: npm ci --force
       # Cache node_modules across different internal jobs in the workflow and across different circleci runs
```
Comment (sreenara) [inline .circleci/config.yml:56]: Are we going to change this to force?

File: packages/node_modules/@webex/react-component-adaptive-card/src/adaptiveCard.scss [hunk 5]
```diff
@@ -45,7 +45,7 @@
             margin: auto 0px auto 6px;
           }
           &.expanded {
-            @extend .md-button:active;
+            @extend .md-button, :active;
```
Comment (sreenara) [inline packages/node_modules/@webex/react-component-adaptive-card/src/adaptiveCard.scss:48]: Is this change needed?

File: scripts/webpack/webpack.prod.babel.js [hunk 6]
```diff
@@ -12,7 +12,10 @@ const HtmlWebpackPlugin = require('html-webpack-plugin');
 const webpackBaseConfig = require('./webpack.base.babel');
 
 const plugins = [
-  new webpack.IgnorePlugin(/^\.\/locale$/, /moment$/)
+  new webpack.IgnorePlugin({
+    resourceRegExp: /^\.\/locale$/,
```
Comment (sreenara) [inline scripts/webpack/webpack.prod.babel.js:16]: was this a new requirement in webpack?

File: scripts/webpack/webpack.base.babel.js [hunk 7]
```diff
@@ -15,20 +15,24 @@ process.env.REACT_WEBEX_VERSION = version;
...
+    new webpack.EnvironmentPlugin({
+      NODE_ENV: process.env.NODE_ENV || 'development',
+      WEBEX_CLIENT_ID: process.env.WEBEX_CLIENT_ID || '',
+      REACT_WEBEX_VERSION: process.env.REACT_WEBEX_VERSION,
+      WDM_SERVICE_URL: process.env.WDM_SERVICE_URL || 'https://wdm-a.wbx2.com/wdm/api/v1',
+      IDBROKER_BASE_URL: process.env.IDBROKER_BASE_URL || 'https://idbroker.webex.com',
+      WEBEX_TEST_USERS_CONVERSATION_SERVICE_URL: process.env.WEBEX_TEST_USERS_CONVERSATION_SERVICE_URL || 'https://conv-a.wbx2.com/conversation/api/v1',
+      WEBEX_CONVERSATION_DEFAULT_CLUSTER: process.env.WEBEX_CONVERSATION_DEFAULT_CLUSTER || 'urn:TEAM:us-east-1_a:identityLookup',
+      FEDERATION: process.env.FEDERATION || '',
+      U2C_SERVICE_URL: process.env.U2C_SERVICE_URL || 'https://u2c.wbx2.com/u2c/api/v1'
```
Comment (sreenara) [inline scripts/webpack/webpack.base.babel.js:27]: Do we need to have the default values here itself?

File: .circleci/config.yml (hunk 4 above)
Comment (Kesari3008) [reply .circleci/config.yml:56]: It's already changed. Please check the latest change

File: scripts/webpack/webpack.prod.babel.js (hunk 6 above)
Comment (Kesari3008) [reply scripts/webpack/webpack.prod.babel.js:16]: Yes

File: scripts/webpack/webpack.base.babel.js (hunk 7 above)
Comment (Kesari3008) [reply scripts/webpack/webpack.base.babel.js]: This change is reverted since this was causing failure for opening space using SpaceID

**Review by sreenara — APPROVED** [review]
